src/java_bytecode/java_object_factory.cpp

```diff
@@ -324,24 +352,52 @@ code_assignt java_object_factoryt::get_null_assignment(
   return code;
 }
 
-/// Initialises an object tree rooted at `expr`, allocating child objects as
-/// necessary and nondet-initialising their members, or if MUST_UPDATE_IN_PLACE
-/// is set, re-initialising already-allocated objects.
-/// \par parameters: `expr`: pointer-typed lvalue expression to initialise
-/// `target_type`: structure type to initialise, which may not match *expr (for
-///   example, expr might be void*)
-/// `alloc_type`: Allocation type (global, local or dynamic)
-/// `update_in_place`: NO_UPDATE_IN_PLACE: initialise `expr` from scratch
-///   MUST_UPDATE_IN_PLACE: reinitialise an existing object MAY_UPDATE_IN_PLACE:
-///   invalid input
+/// Initializes the pointer-typed lvalue expression `expr` to point to an object
+/// of type `target_type`, recursively nondet-initializing the members of such
+/// object. Code emitted mainly depends on \p update_in_place:
+///
+/// When in NO_UPDATE_IN_PLACE mode, the code emitted looks like:
+///
+///   struct new_object obj; // depends on alloc_type
+///   <expr> := &obj
+///   // recursive initialization of obj in NO_UPDATE_IN_PLACE mode
+///
+/// When in MUST_UPDATE_IN_PLACE mode, all code is emitted by a recursive call
+/// to gen_nondet_init in MUST_UPDATE_IN_PLACE mode, and looks like:
```

Comment:
I kindly partially disagree :)

It's indeed an implementation detail. But the existence of this method is also an implementation detail of the parent class and I believe that such implementation detail is more than relevant to clarify what it does.

Notice that there are two similarly named methods, `gen_nondet_pointer_init` and `gen_pointer_target_init`, and a message of the kind you propose will IMHO fail to clarify who does what.